src/core.cpp

```diff
@@ -39,6 +41,7 @@ EXTERN int luasteam_shutdown(lua_State *L) {
     luasteam::shutdown_friends(L);
     luasteam::shutdown_common(L);
     luasteam::shutdown_apps(L);
+    luasteam::shutdown_input(L);
```

Comment:
I see it's not true for `shutdown_apps`, but it's usually better to have shutdown in the opposite order of init. Can you change `input` (and `apps`) so this is true?

Reply:
This also wasn't addressed. Since init ends with `init_apps` and then `init_input`, can you make shutdown begin with `shutdown_input` and then `shutdown_apps`?